lib/std/debug.zig

```diff
@@ -27,185 +42,475 @@ pub const runtime_safety = switch (builtin.mode) {
+    }
+
+    // We define some of the bulkier functions outside of this namespace to
+    // avoid clutter.
+
+    /// Formats a single line for a stack trace.
+    pub const formatStackTraceLine = defaultFormatStackTraceLine;
+
+    /// Loads a stack trace into the provided argument.
+    pub const captureStackTraceFrom = defaultCaptureStackTraceFrom;
+
+    /// This is the termination function for panic. It must be `noreturn`.
+    ///
+    /// When overriding panicTerminate, if you *must* lock a mutex, be careful
+    /// not to panic!  Otherwise, this will would call panicTerminate
+    /// recursively and deadlock.
+    pub const panicTerminate = os.abort;
+};
+
+const is_root_config = @hasDecl(root, "debug_config");
+const config = if (is_root_config)
+    root.debug_config
+else
+    default_config;
+
+const has_writer_decl = @hasDecl(config, "getWriter");
+
+/// Slightly different name than in config to avoid redefinition in default.
+const getWrit = if (has_writer_decl)
+    config.getWriter
+else
+    default_config.getWriter;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const getTTYConfig = if (@hasDecl(config, "detectTTYConfig"))
+    config.detectTTYConfig
+else if (has_writer_decl)
+    @compileError("getWriter exists in config, so detectTTYConfig must also exist")
+else
+    default_config.detectTTYConfig;
+
+const has_symbol_map_decl = @hasDecl(config, "SymbolMap");
+const using_default_symbol_map = !(is_root_config and has_symbol_map_decl);
+
+/// Slightly different name than in config to avoid redefinition in default.
+const SymMap = if (has_symbol_map_decl)
+    config.SymbolMap
+else
+    default_config.SymbolMap;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const writeLineFromSourceFile = if (@hasDecl(config, "attemptWriteLineFromSourceFile"))
+    config.attemptWriteLineFromSourceFile
+else
+    default_config.attemptWriteLineFromSourceFile;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const fmtStackTraceLine = if (@hasDecl(config, "formatStackTraceLine"))
+    config.formatStackTraceLine
+else
+    default_config.formatStackTraceLine;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const capStackTraceFrom = if (@hasDecl(config, "captureStackTraceFrom"))
+    config.captureStackTraceFrom
+else
+    default_config.captureStackTraceFrom;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const getPanicStageTL = if (@hasDecl(config, "getPanicStage"))
+    config.getPanicStage
+else
+    default_config.getPanicStage;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const sleepForev = if (@hasDecl(config, "sleepForever"))
+    config.sleepForever
+else
+    default_config.sleepForever;
+
+/// Slightly different name than in config to avoid redefinition in default.
+const panicTerm = if (@hasDecl(config, "panicTerminate"))
```

Comment:
See how much noise you're adding with those switchable definitions?

Reply:
If it would be better, this could be done generically with metaprogramming. I just thought that would be worse.